### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by jeremi**

### CODE REVIEW

**Review by shibu-narayanan — COMMENTED** [review]
Many areas are commented. Further changes are expected subsequently? How soon do you think that todo's can be removed?

**Comment by jeremi** [comment]
@shibu-narayanan I cleaned up th commented code. As for the TODO, yes, they will need to be solved, but some need some time to think about. We should create tickets for them in Jira.

**Comment by shibu-narayanan** [comment]
> @shibu-narayanan I cleaned up th commented code. As for the TODO, yes, they will need to be solved, but some need some time to think about. We should create tickets for them in Jira.

Yes, makes sense. Puneet is actively setting up things. Let's open up JIRA as soon as its available.

File: g2p_programs/models/managers/cycle_manager.py [hunk 1]
```diff
@@ -233,14 +233,55 @@ def mark_ended(self, cycle):
...
+            }
+        )
+
+        jobs = []
+        max_jobs_per_batch = 100
+        ctr = 0
+        entitlements = []
+        max_rec = len(entitlement_ids)
+        for ctr_entitlements, entitlement in enumerate(entitlement_ids, 1):
+            ctr += 1
```
Comment (jeremi) [inline g2p_programs/models/managers/cycle_manager.py]: You don't need a new variable for this you could use `ctr_entitlements % max_jobs_per_batch`

File: g2p_programs/models/managers/cycle_manager.py [hunk 2]
```diff
@@ -233,14 +233,55 @@ def mark_ended(self, cycle):
...
+                entitlements_ids = self.env["g2p.entitlement"].search(
+                    [("id", "in", entitlements)]
+                )
+                jobs.append(self.delayable()._validate_entitlements(entitlements_ids))
+                ctr = 0
+                entitlements = []
+
+        main_job = group(*jobs)
+        main_job.on_done(
+            self.delayable().mark_import_as_done(cycle, _("Entitlement Ready."))
```
Comment (jeremi) [inline g2p_programs/models/managers/cycle_manager.py]: Let's rename it to `Entitlement approved.`

**Review by jeremi — COMMENTED** [review]
@emjay0921 can you do this changes?

**Review by gonzalesedwin1123 — APPROVED** [review]

**Review by shibu-narayanan — APPROVED** [review]

File: g2p_programs/models/managers/cycle_manager.py (hunk 1 above)
Comment (emjay0921) [reply g2p_programs/models/managers/cycle_manager.py]: Done @jeremi

File: g2p_programs/models/managers/cycle_manager.py (hunk 2 above)
Comment (emjay0921) [reply g2p_programs/models/managers/cycle_manager.py]: Done @jeremi

**Pull request merged**